added peername to socket stats

bookshelfdave opened the pull request:
Suggestions for riak_repl_util: peername error text? It must work in JSON.

Commit:
added peername to socket stats

jonmeredith commented on src/riak_repl2_fscoordinator_serv.erl:
```diff
@@ -171,4 +172,4 @@ get_matching_address(Node, NormIP, Masked) ->
 peername_to_string({{A,B,C,D},Port}) ->
-    lists:flatten(io_lib:format("~B.~B.~B.~B:~B", [A,B,C,D,Port])).
\ No newline at end of file
+    lists:flatten(io_lib:format("~B.~B.~B.~B:~B", [A,B,C,D,Port])).
```
I'd go for

```
inet_parse:ntoa(IpTuple) ++ [$: | integer_to_list(Port)].
```

in case we ever go IPv6 - but it's a bit frustrating for formatting with extra braces around it  `[x:x:x:x:x:x:x:x]:p`

jonmeredith commented:
Code reads well, I'll take it for a spin.

Commit:
Added sockname to TCP mon stats and hardened status call.

Extra paranoia around result of calling peername/sockname.  It
will probably return {error, einval} on a closed socket
but it doesn't hurt to take precautions.

jonmeredith commented:
Added a call to sockname too so we can uniquely identify connections.

```
               {socket,
                          [{peername,"192.168.169.140:9087"},
                           {sockname,"192.168.169.31:39104"},
                           {recv_avg,"[78, 76, 76, 76, 76, 76, 76]"},
                           {recv_cnt,"[1, 0, 0, 0, 0, 0]"},
                           {recv_dvi,"[16, 16, 16, 16, 16, 16, 16]"},
                           {recv_kbps,"[0, 0, 0, 0, 0, 0]"},
                           {recv_max,"[90, 90, 90, 90, 90, 90, 90]"},
                           {send_cnt,"[1, 0, 0, 0, 0, 0]"},
                           {send_kbps,"[0, 0, 0, 0, 0, 0]"},
                           {send_pend,"[0, 0, 0, 0, 0, 0, 0]"}]}]}]
```

Commit:
fix error messages for peername/sockname
